HPXMLtoOpenStudio/resources/hvac_sizing.rb

```diff
@@ -1960,6 +1962,127 @@ def self.apply_hvac_fixed_capacities(hvac_sizing_values, hvac_heating, hvac_cool
     end
   end
 
+  def self.apply_hvac_fixed_airflows(hvac_sizing_values, hvac_heating, hvac_cooling)
+    hvac_sizing_values.Heat_Airflow_isdefaulted = true
+    hvac_sizing_values.Cool_Airflow_isdefaulted = true
+
+    if not hvac_cooling.nil?
+      fixed_cooling_airflow = hvac_cooling.cooling_airflow_cfm
+      max_cooling_airflow = hvac_cooling.max_cooling_airflow_cfm
+    end
+    if (not fixed_cooling_airflow.nil?) && (hvac_sizing_values.Cool_Airflow > 0) && max_cooling_airflow.nil?
+      hvac_sizing_values.Cool_Airflow = fixed_cooling_airflow
+      hvac_sizing_values.Cool_Airflow_isdefaulted = false
+    end
+    if not hvac_heating.nil?
+      fixed_heating_airflow = hvac_heating.heating_airflow_cfm
+      max_heating_airflow = hvac_heating.max_heating_airflow_cfm
+    end
+    if (not fixed_heating_airflow.nil?) && (hvac_sizing_values.Heat_Airflow > 0) && max_heating_airflow.nil?
+      hvac_sizing_values.Heat_Airflow = fixed_heating_airflow
+      hvac_sizing_values.Heat_Airflow_isdefaulted = false
+    end
+  end
+
+  def self.apply_hvac_airflow_adjustments(mj, hvac_sizing_values, hvac_heating, hvac_cooling)
+    '''
+    Airflow Adjustments
+    '''
+
+    # Maximum airflow rate allowed for the duct system
+    max_airflow_allowed = get_max_airflow(hvac_heating, hvac_cooling)
+    return if max_airflow_allowed.nil?
+
+    # W/cfm of the blower fan
+    fan_watts_per_cfm = get_fan_watts_per_cfm(hvac_heating, hvac_cooling)
+    return if fan_watts_per_cfm.nil?
+
+    # Maximum autosized airflow
+    autosized_heating_airflow = hvac_sizing_values.Heat_Airflow
+    autosized_cooling_airflow = hvac_sizing_values.Cool_Airflow
+    max_airflow = [autosized_heating_airflow, autosized_cooling_airflow].max
+
+    # If maximum autosized airflow is not limited, do nothing
+    return if max_airflow <= max_airflow_allowed
+
+    # Adjust heating or cooling system size for the upgraded buildings based on max allowed airflow rate
+    if hvac_sizing_values.Cool_Airflow > hvac_sizing_values.Heat_Airflow
+      # Design cooling airflow rate exceeds max allowed, adjust cooling capacity
+      hvac_sizing_values.Cool_Airflow = max_airflow_allowed
+      hvac_sizing_values.Cool_Capacity *= hvac_sizing_values.Cool_Airflow / autosized_cooling_airflow
+      hvac_sizing_values.Cool_Capacity_Sens = hvac_sizing_values.Cool_Capacity * @hvac_cooling_shr
+      hvac_sizing_values.Cool_Airflow_isdefaulted = false
+
+      if hvac_sizing_values.Heat_Capacity > 0
+        supply_air_temp = hvac_heating.additional_properties.supply_air_temp
+
+        hvac_sizing_values.Heat_Capacity = hvac_sizing_values.Cool_Capacity
+        hvac_sizing_values.Heat_Airflow = calc_airflow_rate_manual_s(mj, hvac_sizing_values.Heat_Capacity, (supply_air_temp - mj.heat_setpoint), hvac_sizing_values.Heat_Capacity)
```

Comment:
There's something about this and the use of `calc_airflow_rate_manual_s` that leads to unexpected results. Same for below. See notes about default tests' expected values around this.